part/part_test.go

```diff
@@ -0,0 +1,544 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package part
+
+import (
+	"encoding/binary"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_commonPrefix(t *testing.T) {
+	check := func(a, b, common string) {
+		actual := string(commonPrefix([]byte(a), []byte(b)))
+		if actual != common {
+			t.Fatalf("expected commonPrefix(%q, %q) to equal %q, but got %q",
+				a, b, common, actual)
+		}
+	}
+
+	check("", "", "")
+	check("", "a", "")
+	check("a", "", "")
+	check("a", "a", "a")
+	check("a", "b", "")
+	check("abc", "d", "")
+	check("d", "abc", "")
+	check("ab", "abc", "ab")
+	check("abc", "ab", "ab")
+}
+
+func Test_search(t *testing.T) {
+	tree := New[[]byte](RootOnlyWatch)
+	_, _, tree = tree.Insert([]byte("a"), []byte("a"))
+	_, _, tree = tree.Insert([]byte("ba"), []byte("ba"))
+	_, _, tree = tree.Insert([]byte("bb"), []byte("bb"))
+	_, _, tree = tree.Insert([]byte("c"), []byte("c"))
+	_, _, tree = tree.Insert([]byte("ca"), []byte("ca"))
+
+	v, _, ok := tree.Get([]byte("nope"))
+	if ok {
+		t.Fatalf("found unexpected value: %v", v)
+	}
+
+	for _, key := range []string{"a", "ba", "bb"} {
+		v, _, ok = tree.Get([]byte(key))
+		if !ok || string(v) != key {
+			t.Fatalf("%q not found (%v) or mismatch %q", key, ok, v)
+		}
+	}
+}
+
+func intKey(n uint64) []byte {
+	return binary.BigEndian.AppendUint64(nil, n)
+}
+
+func Test_simple_delete(t *testing.T) {
+	tree := New[uint64]()
+	txn := tree.Txn()
+
+	_, hadOld := txn.Insert(intKey(1), 1)
+	require.False(t, hadOld)
+
+	_, hadOld = txn.Insert(intKey(2), 2)
+	require.False(t, hadOld)
+
+	_, hadOld = txn.Delete(intKey(1))
+	require.True(t, hadOld)
+
+	_, _, ok := txn.Get(intKey(1))
+	require.False(t, ok)
+}
+
+func Test_delete(t *testing.T) {
+	tree := New[uint64]()
+
+	// Do multiple rounds with the same tree.
+	for round := 0; round < 10; round++ {
+		// Use a random amount of keys in random order to exercise different
```

Comment:
can we make this either print the seed or otherwise make it easy to repro the specific random sequence which fails?